SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialNetDriver.cpp

```diff
@@ -191,7 +191,19 @@ void USpatialNetDriver::OnConnectedToSpatialOS()
 
 void USpatialNetDriver::CreateAndInitializeCoreClasses()
 {
-	SpatialOutputDevice = MakeUnique<FSpatialOutputDevice>(Connection, TEXT("Unreal"));
+	int32 PIEIndex = -1; // -1 is Unreal's default index when not using PIE
+#if WITH_EDITOR
+	if (IsServer())
+	{
+		PIEIndex = GEngine->GetWorldContextFromWorldChecked(GetWorld()).PIEInstance;
```

Comment:
How about we don't even create a SpatialOutputDevice if `GPlayInEditorID != PIEIndex`.

Also, I liked how this function specifically created a bunch of objects and then initialized them. I would argue that the OutputDevice isn't "core" either so what if we moved it to its own function which we called?